### ISSUE TITLE
removing children when both parents dead

### ISSUE THREAD

**Issue by r3wt** [algernon-A/Lifecycle-Rebalance-Revisited#22]
https://github.com/algernon-A/Lifecycle-Rebalance-Revisited/blob/master/Code/Patches/CitizenManagerPatches.cs#L100

This is what causes death waves to become inevitable with this mod. when parents die, the children are removed, and thus there is a likelihood you will end up with a city with no children, lots of seniors who are dying. Can you provide any documentation for the data structures and method of the Citizen Manager and Citizen classes? I am wondering if possible to "adopt" the children into another family. Also is there any developers guide for this mod?

**Comment by algernon-A** [comment]
That's incorrect, and is actually counter to the function.  Households where both parents have died become 'zombie' households, historically preventing the CitizenUnit upgrade tick and thus having the children never age up, while also preventing other citzens from moving into the unit; hence why this mod introduced the orphan removal many versions ago.  There is also now an orphan center mod to also deal with this issue for those who don't use this mod.

The orphan removal only works when BOTH parents die and there are still children present to prevent this.  
As children will attempt to move out once they reach young adult status, and will move out absolutely when they become adults, it is only a small minority of cases that need to be cleaned up like this (unless you already have significantly elevated mortality rates, in which case it won't make a practical difference on top of that).  When orphans are removed, the unit is cleared for a new family to move in, making proper use of the unit with a new generational difference.

As with most things in the game, there is no documentation beyond the base API, and certainly no documentation of the data structures and methods.  You have to analyse and reverse-engineer the game code (difficult at first, but does become easier with experience).  The game code is unencrypted and unobfuscated, and can be easily read with an ILCode decompiler (I personally use ILSpy, but there are others).

**Comment by algernon-A** [comment]
Oh, additionally: as you've no doubt seen from the code, all such orphan removals are logged, so enable the detailed logging option to see the actual volume/impact of orphan removals in-game.
